### PR TITLE
Issue 4235

### PR DESCRIPTION
**Pull request by herbdool**
Addresses https://github.com/backdrop/backdrop-issues/issues/4235

### CODE REVIEW

**Comment by herbdool** [comment]
Failing test is random.

File: core/modules/field/modules/list/list.module [hunk 1]
```diff
@@ -519,7 +530,130 @@ function list_field_formatter_view($entity_type, $entity, $field, $instance, $la
...
+        '#default_value' => $settings['format'],
+      );
+
+      $element['custom_on'] = array(
+        '#type' => 'textfield',
+        '#title' => t('Custom output for On'),
+        '#default_value' => $settings['custom_on'],
+        '#states' => array(
+          'visible' => array(
+            'select[name="fields[' . $field['field_name'] . '][settings_edit_form][settings][format]"],select[name="options[settings][format]"]' => array('value' => 'custom'),
```
Comment (klonos) [inline core/modules/field/modules/list/list.module]: Can we please split these 2 conditionals, to make it more readable? Like so:

```suggestion
            array(
              array('select[name="fields[' . $field['field_name'] . '][settings_edit_form][settings][format]"]' => array('value' => 'custom')),
              'or',
              array('select[name="options[settings][format]"]' => array('value' => 'custom')),
            ),
```

File: core/modules/field/modules/list/list.module [hunk 2]
```diff
@@ -519,7 +530,130 @@ function list_field_formatter_view($entity_type, $entity, $field, $instance, $la
...
+        ),
+      );
+
+      $element['custom_off'] = array(
+        '#type' => 'textfield',
+        '#title' => t('Custom output for Off'),
+        '#default_value' => $settings['custom_off'],
+        '#states' => array(
+          'visible' => array(
+            'select[name="fields[' . $field['field_name'] . '][settings_edit_form][settings][format]"],select[name="options[settings][format]"]' => array('value' => 'custom'),
```
Comment (klonos) [inline core/modules/field/modules/list/list.module]: ...same as previous comment:

```suggestion
            array(
              array('select[name="fields[' . $field['field_name'] . '][settings_edit_form][settings][format]"]' => array('value' => 'custom')),
              'or',
              array('select[name="options[settings][format]"]' => array('value' => 'custom')),
            ),
```

**Review by klonos — APPROVED** [review]
LGTM now 👍

File: core/modules/field/modules/list/tests/list.test [hunk 3]
```diff
@@ -436,3 +436,109 @@ class ListFieldUITestCase extends FieldTestCase {
...
+    // Enable the default setting to use custom reverse on/off text.
+    $instance_info = field_info_instance('node', $this->field_name, $this->type);
+    $instance_info['display']['default']['settings']['reverse'] = TRUE;
+    field_update_instance($instance_info);
+
+    $this->backdropGet('node/' . $node->nid);
+    $this->assertText($off, format_string('Custom reversed on value is present.'));
+  }
+
+}
```
Comment (jenlampton) [inline core/modules/field/modules/list/tests/list.test]: Missing line break at end of file.

File: core/modules/field/modules/list/tests/list.tests.info [hunk 4]
```diff
@@ -15,3 +15,9 @@ name = List field UI
 description = Test the List field UI functionality.
 group = Field types
 file = list.test
+
+[ListFieldDisplayTestCase]
+name = List field Display
+description = Test the List field Display formatting.
+group = Field types
+file = list.test
```
Comment (jenlampton) [inline core/modules/field/modules/list/tests/list.tests.info]: Missing line break at end of file here too

File: core/modules/field/modules/list/tests/list.test (hunk 3 above)
Comment (quicksketch) [inline core/modules/field/modules/list/tests/list.test]:
```suggestion
}

```

**Comment by quicksketch** [comment]
Thanks! This looks great! I merged into 1.x for 1.15.0.

**Pull request merged**